src/components/Profile/Settings.js

```diff
@@ -0,0 +1,118 @@
+/* eslint-disable react/destructuring-assignment */
+import React, { Component } from 'react';
+import { Form } from 'semantic-ui-react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+// import firebase
+import firebase from '../../firebase/config';
+
+// import action creator for update profile
+import {
+  updateProfileRequest,
+} from '../../store/actions/profileActions';
+
+// import Button from './Button';
+import Button from '../Button';
+
+// import style
+import './Setting.scss';
+
+class Settings extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      firstname: '',
+      lastname: '',
+      bio: '',
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    const userData = { profile: this.state };
+    const { username } = this.props;
+    return this.props.updateProfileRequest(userData, username);
+  };
+
+  handleChange(event) {
+    this.setState({ [event.target.name]: event.target.value });
+  }
+
+  render() {
+    const {
+      firstname, lastname, bio
+    } = this.state;
+    return (
+      <div className="settings">
+        <div className="edit-profile">
+          <Form onSubmit={this.handleSubmit}>
+            <Form.Group widths="equal">
+              <Form.Input
+                fluid
+                label="First name"
+                placeholder="John"
+                name="firstname"
+                defaultValue={firstname}
+                onChange={this.handleChange}
+              />
+              <Form.Input
+                fluid
+                label="Last name"
+                placeholder="Doe"
+                name="lastname"
+                defaultValue={lastname}
+                onChange={this.handleChange}
+              />
+            </Form.Group>
+            <Form.TextArea
+              label="Bio"
+              name="bio"
+              className="textareastyled"
+              placeholder="Talk about yourself."
+              maxLength="300"
+              defaultValue={bio}
+              onChange={this.handleChange}
+              rows={5}
+            />
+            <div
+              className="field"
+            >
+
+              <Button
+                type="submit"
+                id="editProfile__button"
+                btnClass="edit-btn"
+                btnName="Edit Profile"
+                btnEvent={this.handleSubmit}
+              />
+            </div>
+          </Form>
+        </div>
+      </div>
+    );
+  }
+}
+
+Settings.propTypes = {
+  updateProfileRequest: PropTypes.func.isRequired,
+  // eslint-disable-next-line react/require-default-props
```

Comment:
Please don't suppress this es-lint warning. Default props are useful for when the component loads the first time.
Simply provide the default props.
`
Settings.defaultProps = {
        updateProfileRequest: () => {},
userData: PropTypes.shape({
    firstname: '',
    lastname: '',
    bio: '',
  }),
username: '',
}

Reply:
i have made the changes